tests/integration/conftest.py

```diff
@@ -253,18 +287,27 @@ async def kubernetes_cluster(request: pytest.FixtureRequest, ops_test: OpsTest):
     bundle_marker = request.node.get_closest_marker("bundle_file")
     if bundle_marker:
         bundle_file = bundle_marker.args[0]
+    bundle_path = Path(__file__).parent / "data" / bundle_file
+    model = "main"
+
+    with ops_test.model_context(model) as the_model:
+        if await is_deployed(the_model, bundle_path):
```

Comment:
Will this fail if the tests failed in the clustering tests?
e.g. https://github.com/canonical/k8s-operator/blob/main/tests/integration/test_k8s.py#L104

Let's say the leader was removed [here](https://github.com/canonical/k8s-operator/blob/main/tests/integration/test_k8s.py#L108) but the next step failed. If the tests are rerun, the `is_deployed` check will fail because the number of expected units is wrong and the bundle is freshly deployed, right?

Reply:
That's mostly correct.  If the `is_deployed` check fails, it will apply the bundle to the model as if the model was empty

This fixture `kubernetes_cluster` has `module` scope meaning that it's run once for each python test module.  In the case of test_k8s -- if the fixture completes successfully once, then each test uses the model that fixture yields.  this `is_deployed` won't be checked between tests runs within that file. 

But if say the tests FAIL, and pytest ends, then a user re-runs the tests again -- this fixture would catch there were not enough `k8s` units, and deploy again from the bundle which should yield the right number of k8s units.

Reply:
makes sense - thanks!